### PR TITLE
qa: use Remote.run_unit_test and ValgrindScanner

### PR DESCRIPTION
**Pull request by VallariAg**
1. add "unit_test_scan" optional config uses `Remote.run_unit_test()`
to scan xml files generated by unit tests to throw better failure
messages (for s3tests and gtests run by workunit).
It also creates "unit_test_summary.yaml" for more exception details
from xml files.
2. previously, valgrind_post() func used grep to find error
from valgrind logs.
now, it uses `ValgrindScanner` to log better exceptions with
traceback and exception kind, along with creating a more detailed
summary in valgrind.yaml in archive.

teuthology PR: https://github.com/ceph/teuthology/pull/1792

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by amathuria** [comment]
@VallariAg This looks great! Could you maybe add some sample tests that were run with this enabled so people can see how it looks?

**Comment by VallariAg** [comment]
@amathuria I'm running the tests again, I'll add the links as soon as they're done!

**Comment by kamoltat** [comment]
@alimaredia RGW change looks okay to you?

**Comment by VallariAg** [comment]
results for:
s3tests: https://pulpito.ceph.com/vallariag-2023-10-26_11:22:52-smoke-main-distro-default-smithi/
valgrind: https://pulpito.ceph.com/vallariag-2023-10-26_12:18:51-rados-main-distro-default-smithi/

an old result for gtests (need to rerun gtests once again to double check if everything is okay, will edit once done): https://pulpito.ceph.com/vallariag-2023-09-21_11:10:03-rados:monthrash-wip-vallariag-testing-distro-default-smithi/

**Comment by alimaredia** [comment]
@VallariAg it's great to be able to quickly see the issues with a failed job. One thing I was wondering about is that the only failure I saw in s3-tests was the first failure, but I saw 4 other test failing. Is there any way to say in the output that there were "4 other failures"?

@cbodley @mkogan1 any thoughts on the valgrind changes?

**Comment by VallariAg** [comment]
@alimaredia here's a run with "total x failed" before the first error message: https://pulpito.ceph.com/vallariag-2023-10-27_11:36:23-smoke-main-distro-default-smithi/
There's also `unit_test_summary.yaml` in archive with failure details of all test cases. 
Let me know if I should add anything else!

**Comment by alimaredia** [comment]
That yaml file with the summary is great and the log output looks good! Is the existence of `unit_test_summary.yaml` going to be documented anywhere?

**Comment by VallariAg** [comment]
@alimaredia It's not documented anywhere right now. Should I add a comment in the python or yaml files? Or is there anyplace appropriate in docs where I can mention about it?

**Comment by jdurgin** [comment]
Perhaps it could be included in this area of the docs describing teuthology archives: https://docs.ceph.com/en/reef/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow/#teuthology-archives

**Comment by VallariAg** [comment]
@alimaredia Please let me know if there are any other changes needed on this PR!

**Comment by VallariAg** [comment]
@alimaredia `ValgrindScanner` is a new feature going to be introduced in teuthology via https://github.com/ceph/teuthology/pull/1792
It's ready to be merged - we were just waiting for all comments/changes on output in this PR.

**Comment by VallariAg** [comment]
jenkins test make check arm64

File: doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow.rst
Comment (zdover23) [inline doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow.rst:1]: https://github.com/ceph/ceph/pull/55982

**Pull request merged**